test/use-checkout-pricing.test.js

```diff
@@ -0,0 +1,154 @@
+import React, { useEffect } from 'react';
+import { render, wait, cleanup } from './support/custom-render';
+import { useCheckoutPricing } from '../lib';
+
+const subscriptionPricingReturn = {
+  plan: jest.fn(() => subscriptionPricingReturn),
```

Comment:
I'm wondering if these methods to use something like `mockResolvedValue` (https://jestjs.io/docs/en/mock-function-api#mockfnmockresolvedvaluevalue) or return promises. Honestly not sure.

Reply:
Looking at it closer, that's definitely the way to go. The `subscriptionPricingReturn` and `checkoutPricingReturn` mocks aren't exactly close to the real implementations.

`recurly.Pricing.Checkout` will return a [`CheckoutPricing`](https://github.com/recurly/recurly-js/blob/master/lib/recurly/pricing/checkout/index.js), which itself is not a `Promise`. The methods on that class will return Promises, but we should not expect the class itself to implement `then`, `catch`, and `finally`.

So the following will behave more closely like the real thing:

```js
const subscriptionPricingReturn = {
  plan: jest.fn().mockResolvedValue(subscriptionPricingReturn),
  addon: jest.fn().mockResolvedValue(subscriptionPricingReturn),
  tax: jest.fn().mockResolvedValue(subscriptionPricingReturn)
};
```

The mock method resolvers will then mock the promise methods appropriately.